app/uk/gov/hmrc/cgtpropertydisposals/service/TaxEnrolmentService.scala

```diff
@@ -72,4 +78,22 @@ class TaxEnrolmentServiceImpl @Inject()(
       case other => Left(Error(s"Received error response from tax enrolment service with http status: $other"))
     }
 
+  override def hasCgtEnrolment(
+    ggCredId: String
+  )(implicit hc: HeaderCarrier): EitherT[Future, Error, Option[TaxEnrolmentRequest]] =
+    for {
+      maybeEnrolmentRequest <- taxEnrolmentRepository
+                                .get(ggCredId)
+                                .leftMap(error => Error(s"Could not check existence of enrolment in mongo: $error"))
+      taxEnrolmentRequest <- EitherT
+                              .fromOption[Future](maybeEnrolmentRequest, Error("No enrolment request found in mongo"))
+      httpResponse <- makeTaxEnrolmentCall(taxEnrolmentRequest)
+      _ <- EitherT
+            .fromEither[Future](handleTaxEnrolmentResponse(httpResponse))
+            .leftMap(error => Error(s"Could not allocate enrolment: $error"))
+      _ <- taxEnrolmentRepository
+            .delete(ggCredId)
+            .leftMap(error => Error(s"Could not delete enrolment request in mongo: $error"))
```

Comment:
Is this correct? Should we return an error if the delete failed? I have asked the GGW team if the ES8 API is idempotent

Reply:
This doesn't matter anymore. If the enrolment did occur but we are unable to delete then the call to this new endpoint will not be made and therefore duplicate tax enrolment calls will not be made.